xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/email/NotificationEmailUserPreferenceManager.java

```diff
@@ -40,4 +42,31 @@
      * @return the diff type configured for the given user
      */
     NotificationEmailDiffType getDiffType(String userId);
+
+    /**
+     * @param userReference reference of a user
+     * @return the diff type configured for the given user
+     * @since 14.4.2
+     */
+    NotificationEmailDiffType getDiffType(DocumentReference userReference);
```

Comment:
Would be better in new APIs to rely on UserReference.

Reply:
I committed a new version using UserReference and with the `default` keyword for new methods but the PR still requires some updates.